src/raw/mod.rs

```diff
@@ -208,17 +208,19 @@ fn bucket_mask_to_capacity(bucket_mask: usize) -> usize {
     }
 }
 
-// Returns a Layout which describes the allocation required for a hash table,
-// and the offset of the buckets in the allocation.
+/// Returns a Layout which describes the allocation required for a hash table,
+/// and the offset of the control bytes in the allocation.
+/// (the offset is also one past last element of buckets)
 ///
 /// Returns `None` if an overflow occurs.
 #[cfg_attr(feature = "inline-more", inline)]
 #[cfg(feature = "nightly")]
 fn calculate_layout<T>(buckets: usize) -> Option<(Layout, usize)> {
     debug_assert!(buckets.is_power_of_two());
 
+    let common_align = usize::max(mem::align_of::<T>(), Group::WIDTH);
     // Array of buckets
-    let data = Layout::array::<T>(buckets).ok()?;
+    let padded_data = Layout::array::<T>(buckets).ok()?.align_to(common_align).ok()?.pad_to_align();
```

Comment:
I don't think this is actually required. `data` is already guaranteeed to be properly aligned for the data and `extend` will ensure that `ctrl_offset` is properly aligned for `Group::WIDTH`.

Reply:
oh I overlooked this.